client/extensions/woocommerce/state/wc-api/products/actions.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Internal dependencies
+ */
+import wp from 'lib/wp';
+import { error } from '../actions';
+import {
+	WOOCOMMERCE_API_CREATE_PRODUCT,
+	WOOCOMMERCE_API_CREATE_PRODUCT_SUCCESS,
+} from '../../action-types';
+
+export function createProduct( siteId, product ) {
+	return ( dispatch ) => {
+		const createAction = {
+			type: WOOCOMMERCE_API_CREATE_PRODUCT,
+			payload: { siteId, product },
+		};
+
+		dispatch( createAction );
+
+		// Filter out any id we might have.
+		const { id, ...productData } = product;
+
+		if ( typeof id === 'number' ) {
+			dispatch( error( siteId, createAction, {
+				message: 'Attempting to create a product which already has a valid id.',
+				product,
+			} ) );
+			return;
+		}
+
+		const jetpackProps = { path: `/jetpack-blogs/${ siteId }/rest-api/` };
+		const httpProps = {
+			path: '/wc/v2/products',
+			body: JSON.stringify( productData ),
+			json: true,
+		};
+
+		// TODO: Modify this to use the extensions data layer.
+		return wp.req.post( jetpackProps, httpProps )
+			.then( ( { data } ) => {
+				dispatch( createProductSuccess( siteId, data ) );
+			} )
+			.catch( err => {
+				dispatch( error( siteId, createAction, err ) );
+			} );
```

Comment:
pardon my coming into this at the last moment here but did you consider using the HTTP system in the data layer?